pandas/tests/indexes/common.py

```diff
@@ -30,7 +31,7 @@
 class Base:
     """ base class for index sub-class tests """
 
-    _holder = None
+    _holder = None  # type: Optional[Type[Index]]
```

Comment:
Does this need to be `Optional`?

Reply:
IMO we should keep the `Optional` otherwise when we use 3.6 type annotations we'll get
`pandas\tests\indexes\common.py:34: error: Incompatible types in assignment (expression has type "None", variable has type "Type[Index]")`

The Optional should only be excluded where we have added `something = None  # type: ...` for the purpose of type checking.

In this case `_holder` was set to `None` before the type annotation was added.

so when we move to 3.6, we will change to
`_holder: Optional[Type[Index]] = None`
whereas if the initialisation was purely added for the type annotations we would change to just 
`_holder: Type[Index]`

Reply:
Isn't the latter what we actually want here though? OK to change to 36 annotation now on this PR since 35 CI should be dropped within the next few days

Reply:
> Isn't the latter what we actually want here though?

IMO out of scope here

but to answer your question, in general, with  3.6, we could reduce the number of `assert <some_variable> is not None` that are sometimes needed when class attributes are initialised to `None`, but we would need to evaluate on a case by case basis.

> OK to change to 36 annotation now on this PR since 35 CI should be dropped within the next few days

i'm not sure a good reason to block since this is not guaranteed.